pkg/meta/redis_csc.go

```diff
@@ -0,0 +1,288 @@
+//go:build !noredis
+// +build !noredis
+
+/*
+ * JuiceFS, Copyright 2020 Juicedata, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package meta
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+)
+
+// setupClientSideCaching configures Redis client-side caching
+func (m *redisMeta) setupClientSideCaching(expiry time.Duration) error {
+	ctx := Background()
+
+	// Store the expiry setting - 0 means infinite (no expiry)
+	m.clientCacheExpiry = expiry
+
+	// For cluster clients, we need a separate connection for tracking
+	if _, ok := m.rdb.(*redis.ClusterClient); ok {
+		// For cluster mode, we should get the master node for our key
+		_, err := m.rdb.(*redis.ClusterClient).MasterForKey(ctx, m.prefix)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Enable tracking - first disable tracking if already enabled
+	_ = m.rdb.Do(ctx, "CLIENT", "TRACKING", "OFF").Err() // Ignore errors if not previously enabled
+
+	// Always use BCAST mode for simplicity
+	err := m.rdb.Do(ctx, "CLIENT", "TRACKING", "ON", "BCAST").Err()

```

Comment:
BCAST mode may not work when there are many connections (>100). Especially when a single mount point may use tens of connections.